playground/ios/SnapshotTests/StackOptionsTest.m

```diff
@@ -0,0 +1,68 @@
+#import <XCTest/XCTest.h>
+#import "LayoutCreator.h"
+#import "CommandsHandlerCreator.h"
+#import <FBSnapshotTestCase/FBSnapshotTestCase.h>
+
+@interface StackOptionsTest : FBSnapshotTestCase
+@property (nonatomic, strong) RNNCommandsHandler* commandsHandler;
+@property (nonatomic, strong) UIWindow* window;
+@end
+
+@implementation StackOptionsTest
+
+- (void)setUp {
+	[super setUp];
+	_window = [[[UIApplication sharedApplication] delegate] window];
+	_commandsHandler = [CommandsHandlerCreator createWithWindow:_window];
+	self.usesDrawViewHierarchyInRect = YES;
+	
+//	Uncomment next line to record new snapshots
+//	self.recordMode = YES;
+}
+
+- (void)tearDown {
+	[super tearDown];
+	_window.rootViewController = nil;
+}
+
+#define RNNStackFlow(NAME, FIRST, SECOND, ...)\
+[self setRootPushAndPop:FIRST secondComponentOptions:SECOND testName:@#NAME];
+
+- (void)testStack_topBar {
+	RNNStackFlow(opaque background, @{@"topBar": @{@"background": @{@"color": @(0xFFFF00FF)}}}, @{@"topBar": @{@"background": @{@"color": @(0xFFFF0000)}}});
+	RNNStackFlow(tansparent background, @{@"topBar": @{@"background": @{@"color": @(0x00FFFFFF)}}}, @{@"topBar": @{@"background": @{@"color": @(0xFFFF0000)}}});
+	RNNStackFlow(translucent background, @{@"topBar": @{@"background": @{@"translucent": @(1)}}}, @{@"topBar": @{@"background": @{@"translucent": @(0)}}});
+	RNNStackFlow(title change, @{@"topBar": @{@"title": @{@"text": @"First Component"}}}, @{@"topBar": @{@"title": @{@"text": @"Second Component"}}});
+	RNNStackFlow(visibility, @{@"topBar": @{@"visible": @(0)}}, @{@"topBar": @{@"visible": @(1)}});
+	RNNStackFlow(title font, @{@"topBar": @{@"title": @{@"text": @"First Component"}}}, (@{@"topBar": @{@"title": @{@"text": @"Second Component", @"fontFamily": @"Arial", @"color": @(0xFFFF00FF), @"fontSize": @(15)}}}));
+}
+
+- (void)setRootPushAndPop:(NSDictionary *)firstComponentOptions secondComponentOptions:(NSDictionary *)secondComponentOptions testName:(NSString *)testName {
```

Comment:
This function is quite long which makes it hard to understand what it does (even though the name explicitly states it, which is great). Splitting it into three smaller functions, where each function does a single task, can help.